File: 0961-long-pressed-name/0961-long-pressed-name.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
            int i=1 , j=1;
    
    //Checking the first character
    if(name.charAt(0) != typed.charAt(0))
        return false;
    
    
    //Traversing the Strings using two pointer
    while(i<name.length() && j<typed.length()){
        if(name.charAt(i) == typed.charAt(j)){
            i++;
            j++;
        }
        
        else {  
            if(typed.charAt(j) == typed.charAt(j-1))
                j++;
            
            else
                return false;    
        }
    }
    
    
    //Case 1 : i and j both reached end of the String
    if(i == name.length() && j == typed.length())
        return true;
    
    //Case 2 : First word is not yet complete traversed. The second word is complete
    else if(i != name.length() && j == typed.length())
        return false;
    
    //Case 3: First word complete. The second word still has some characters left.
    else{
        while(j != typed.length()){
            if(typed.charAt(j) == typed.charAt(j-1))
                j++;
            
            else
                return false;
        }
    }
    return true;
}
}

File: 0961-long-pressed-name/0961-long-pressed-name.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        int i =0;
        int j =0;
        char c = name[0];

        while(i<name.size() && j<typed.size()){
            int count=0;
            while(name[i]==c){
                count++;
                i++;
            }
            while(typed[j]==c){
                count--;
                j++;
            }

            if(name[i]!=typed[j] || count>0)return false;
            c = name[i];

        }

        return true;

    }
};